### PR TITLE
Added support for Phonetic Matching

### LINKED ISSUE #87: Allow adding PHONETIC Attribute on Create Schema
related to RediSearch/RediSearch#1607

```
sc := redisearch.NewSchema(redisearch.DefaultOptions).
		AddField(redisearch.Field{
		 	Name:     "text",
		 	Type:     redisearch.TextField,
		 	Sortable: true,
		 	Options:  "PHONETIC dm:en",
	       }).
...
```
returns error `Error on TextField serialization`.

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR fixes #87 by adding `PhoneticMatcherType` property to text fields.
It includes a new example to demonstrate phonetic matching features ( example name `ExampleClient_CreateIndexWithIndexDefinition_phonetic`)

### CODE REVIEW

**Review by hualoqueros — APPROVED** [review]
this is looks great.. that solve my phonetic problem

**Review by ashtul — APPROVED** [review]

**Review by ashtul — APPROVED** [review]

File: redisearch/example_schema_test.go [hunk 1]
```diff
@@ -50,3 +51,50 @@ func ExampleCreateIndex_temporary() {
...
+	for info.IsIndexing {
+		time.Sleep(time.Second)
+		info, _ = c.Info()
+	}
+
+	_, total, _ := c.Search(redisearch.NewQuery("Jon").
+		SetReturnFields("name"))
+
+	// Verify that the we've received 2 documents ( Jon and John )
+	fmt.Printf("Total docs replied %d\n", total)
```
Comment (MeirShpilraien) [inline redisearch/example_schema_test.go:97]: Why print and not assert?

File: redisearch/schema_test.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ func TestSerializeSchema(t *testing.T) {
...
 		{"default-and-sortable-text-field", args{NewSchema(DefaultOptions).AddField(NewSortableTextField("text-field", 10)), redis.Args{}}, redis.Args{"SCHEMA", "text-field", "TEXT", "WEIGHT", float32(10.0), "SORTABLE"}, false},
 		{"default-and-text-with-options", args{NewSchema(DefaultOptions).AddField(NewTextFieldOptions("text-field", TextFieldOptions{Weight: 5.0, Sortable: true, NoStem: false, NoIndex: false})), redis.Args{}}, redis.Args{"SCHEMA", "text-field", "TEXT", "WEIGHT", float32(5.0), "SORTABLE"}, false},
+		{"default-and-text-with-phonetic-en", args{NewSchema(DefaultOptions).AddField(NewTextFieldOptions("text-field", TextFieldOptions{PhoneticMatcher: PhoneticDoubleMetaphoneEnglish})), redis.Args{}}, redis.Args{"SCHEMA", "text-field", "TEXT", "PHONETIC", "dm:en"}, false},
+		{"default-and-text-with-phonetic-pt", args{NewSchema(DefaultOptions).AddField(NewTextFieldOptions("text-field", TextFieldOptions{PhoneticMatcher: PhoneticDoubleMetaphonePortuguese})), redis.Args{}}, redis.Args{"SCHEMA", "text-field", "TEXT", "PHONETIC", "dm:pt"}, false},
```
Comment (MeirShpilraien) [inline redisearch/schema_test.go:64]: What those are used for?

**Review by MeirShpilraien — COMMENTED** [review]
Looks good, 2 small comments.

File: redisearch/example_schema_test.go (hunk 1 above)
Comment (filipecosta90) [reply redisearch/example_schema_test.go:97]: @MeirShpilraien this is an example. We're supposed to print and check the output matches. This is ment for godoc:
```
        // Verify that the we've received 2 documents ( Jon and John )
	fmt.Printf("Total docs replied %d\n", total)

	// Output: Total docs replied 2
```
Expected godoc ( check output field):
image

File: redisearch/schema_test.go (hunk 2 above)
Comment (filipecosta90) [reply redisearch/schema_test.go:64]: to test the internal schema serialization.

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**